app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -0,0 +1,240 @@
+require 'azure_mgmt_compute'
+require 'azure_mgmt_resources'
+require 'azure_mgmt_network'
+require 'azure_mgmt_storage'
+
+module ForemanAzureRM
+  class AzureRMCompute
+    attr_accessor :sdk
+    attr_accessor :azure_vm
+
+    Compute = Azure::Compute::Profiles::Latest::Mgmt
+    ComputeModels = Compute::Models
+
+    def initialize(azure_vm: ComputeModels::VirtualMachine.new, sdk: sdk)
+      @azure_vm = azure_vm
+      @sdk = sdk
+    end
+
+    def id
+      @azure_vm.id
+    end
+
+    def name
+      @azure_vm.name
+    end
+
+    def persisted?
+      !!identity && !!id
+    end
+
+    def region
+      @azure_vm.location
+    end
+
+    def region=(setregion)
+      @azure_vm.location = setregion
+    end
+
+    def resource_group
+      @azure_vm.resource_group
+    end
+
+    def resource_group=(setrg)
+      @azure_vm.resource_group = setrg
+    end
+
+    def vm_size
```

Comment:
I suppose the property is here because it is used by Foreman. That means we should initialize a getter for it.

The easiest thing to do would be to use [`delegate`](https://apidock.com/rails/Module/delegate):
``` ruby
class AzureRMCompute
  delegate :platform, :username to: :@azure_vm :allow_nil: true
end
```

Reply:
As these methods are not currently available on `@azure_vm`, `delegate` will give error.
Instead of `delegate`, I think `attr_reader` will work here.

Reply:
Deleted my previous comment as I thought `:platform` method is not able on @azure_vm but it is accessible through other property of @azure_rm.

Reply:
If those getter methods are available on @azure_rm then only `delegate` will work otherwise use `attr_reader`.

Reply:
Turns out it can be accessed using `@azure_vm.storage_profile.os_disk.os_type`. But, since `platform` is our wrapper, how should we approach to delegate it on `:azure_vm`?